mds/apis/agency_api/v0_x/vehicles.py

```diff
@@ -93,7 +96,20 @@ class DeviceRegisterSerializer(serializers.Serializer):
     model = serializers.CharField(required=False, help_text="Vehicle Model.")
 
     def create(self, validated_data):
-        provider_id = self.context["request"].user.provider_id
+        data_broker_id = self.context["request"].user.provider_id
+
+        try:
+            provider_id = str(validated_data.pop("provider_id"))
```

Comment:
```suggestion
            provider_id = str(validated_data.pop("provider_id", data_broker_id))
```
And then no need for a try/except.